lib/crowbar/init/helpers.rb

```diff
@@ -301,6 +303,15 @@ def crowbar_reset
 
         status
       end
+
+      # once the admin server is upgraded and rebooted, we have to set the status of the
+      # admin_upgrade step to 'passed'
+      def end_admin_upgrade_step
+        upgrade_status = ::Crowbar::UpgradeStatus.new
+        return false unless upgrade_status.current_step == :admin_upgrade
```

Comment:
This introduces a race condition - status could change after this check and before `#end_step` is called.